### PR TITLE
Add 'icon' property to Binary Sensor, Climate, Cover, Fan, and Light platforms

### PR DESCRIPTION
**Pull request by paulmonigatti**
## Description:
User-facing changes resulting from esphome/esphome#2418

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2418

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by paulmonigatti** [comment]
This is currently working in Home Assistant via MQTT, however I'll still need to update the HA ESPHome component before it will work via the native API.

I'm thinking it will be easiest to wait until esphome/aioesphomeapi#106 is merged and available on PyPI before getting started with the HA side of things, in which case this documentation should be held back in the meantime. Let me know what works best.

**Review by OttoWinter — APPROVED** [review]
LGTM, thanks 👍

**Pull request merged**
